lib/rubocop/cop/rails/where_missing.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Check for the use `left_joins` and `where` method for finding missing relationship records.
+      #
+      # This cop is enabled in Rails 6.1 or higher.
+      #
+      # @example
+      #   # bad
+      #   Foo.left_joins(:foo).where(foos: { id: nil })
+      #
+      #   # good
+      #   Foo.where.missing(:foo)
+      #
+      class WhereMissing < Base
+        include RangeHelp
+        extend AutoCorrector
+        extend TargetRailsVersion
+
+        MSG = 'Use `where.missing(:%<association>s)` instead of ' \
+              '`%<left_joins_method>s(:%<association>s).where(%<association>ss: { id: nil })`.'
+        RESTRICT_ON_SEND = %i[left_joins left_outer_joins].freeze
+
+        minimum_target_rails_version 6.1
+
+        # @!method where_arguments(node)
+        def_node_search :where_arguments, <<~PATTERN
+          {
+            (send ... :where (hash  (pair $(sym _) (hash (pair (sym :id) (nil)))) ... ))
+            (send ... :where (hash ... (pair $(sym _) (hash (pair (sym :id) (nil)))) ... ))
+          }
+        PATTERN
+
+        # @!method where(node)
+        def_node_search :where, <<~PATTERN
+          {
+            $(send ... :where (hash (pair (sym _) (hash (pair (sym :id) (nil)))) ... ))
+            $(send ... :where (hash ... (pair (sym _) (hash (pair (sym :id) (nil)))) ... ))
+          }
+        PATTERN
+
+        # @!method missing_relationship?(node)
+        def_node_search :missing_relationship?, <<~PATTERN
+          {
+            (pair (sym _) (hash (pair (sym :id) (nil))))
+            (pair (sym _) (hash ... (pair (sym :id) (nil))))
```

Comment:
Again, there is the `<(pair (sym :id) (nil) ...>` to match when one of the pairs matches. Besides you don't have the case of pairs after the `id: nil` (e.g. trailing `...`). I'm mentioning the last as that might indicate a missing test case

Reply:
This was unnecessary for the following cases
```ruby
(pair (sym _) (hash ... (pair (sym :id) (nil))))
```

> Besides you don't have the case of pairs after the id: nil (e.g. trailing ...) .

I believe I have tested it here, but please point out if I am mistaken.
 https://github.com/rubocop/rubocop-rails/blob/4cdcf29b294dcc6853deffc0eba5bfe9c9d191f7/spec/rubocop/cop/rails/where_missing_spec.rb#L38-L58

Reply:
> Besides you don't have the case of pairs after the id: nil (e.g. trailing ...) .

I believe I have tested it here, but please point out if I am mistaken.
 https://github.com/rubocop/rubocop-rails/blob/4cdcf29b294dcc6853deffc0eba5bfe9c9d191f7/spec/rubocop/cop/rails/where_missing_spec.rb#L38-L58